reports/src/main/java/com/ivy/reports/ReportViewModel.kt

```diff
@@ -1,415 +1,418 @@
+                expenses = expense,
+                incomeTransactionsCount = incomeCount,
+                expenseTransactionsCount = expenseCount,
+                treatTransfersAsIncExp = transfersAsIncomeExpense
+            )
+        }.lambda()
+    }
 
-                if (it.description != null && it.description!!.isNotEmpty()) {
-                    includeKeywords.forEach { keyword ->
-                        if (it.description!!.containsLowercase(keyword)) {
-                            return@filter true
-                        }
-                    }
-                }
+    private suspend fun export(context: Context, fileUri: Uri, onShareUI: (Uri) -> Unit) {
+        val filter = stateVal().filter
 
-                false
-            }
-            .filter {
-                //Filter by Excluded Keywords
-
-                val excludedKeywords = filter.excludeKeywords
-                if (excludedKeywords.isEmpty()) return@filter true
-
-                if (it.title != null && it.title!!.isNotEmpty()) {
-                    excludedKeywords.forEach { keyword ->
-                        if (it.title!!.containsLowercase(keyword)) {
-                            return@filter false
-                        }
-                    }
-                }
+        filter ?: return
+        if (!filter.validateFilter()) return
 
-                if (it.description != null && it.description!!.isNotEmpty()) {
-                    excludedKeywords.forEach { keyword ->
-                        if (it.description!!.containsLowercase(keyword)) {
-                            return@filter false
-                        }
-                    }
-                }
+        updateState {
+            it.copy(loading = true)
+        }
 
-                true
+        exportCSVLogic.exportToFile(
+            context = context,
+            fileUri = fileUri,
+            exportScope = {
+                allTransactions.value.toOld()
             }
-            .toList()
+        )
+
+        uiThread {
+            onShareUI(fileUri)
+        }
+
+        updateState {
+            it.copy(loading = false)
+        }
     }
 
-    private fun String.containsLowercase(anotherString: String): Boolean {
-        return this.toLowerCaseLocal().contains(anotherString.toLowerCaseLocal())
+    private suspend fun filterTransactions(
+        baseCurrency: String,
+        filter: ReportFilter
+    ) = {
+        ByTypeIn(filter.trnTypes.toNonEmptyList()) and
+                ByAllDate(filter.period) and
+                filterByAccountIn(filter.accounts) and
+                ByCategoryIn(filter.categories.noneCategoryFix().toNonEmptyList())
+    } then queryTransactionsAct then {
+        filterByAmount(baseCurrency = baseCurrency, filter = filter, transList = it)
+    } then {
+        filterByWords(filter, it)
+    } then { allTrans ->
+        val nonPlannedPaymentsTransactions = allTrans.filter { it.time is TrnTime.Actual }
```

Comment:
`nonPlannedPaymentsTransactions` -> `actualTrns`; I think there's also a function in `:core:fuctions`: to filter actual.